src/io.ts

```diff
@@ -91,15 +91,16 @@ export function readTodoStorageFile(todoStorageFilePath: string): Operation[] {
     encoding: 'utf8',
   });
 
-  let operations = todoContents.split(EOL) as OperationOrConflictLine[];
+  // when splitting by EOL, make sure to filter off the '' caused by the final EOL
+  let operations: OperationOrConflictLine[] = todoContents.split(EOL).filter((op: OperationOrConflictLine | '') => op !== '')
```

Comment:
This is the starting point of the fix.
`todoContents.split(EOL)` results in `['op1', 'op2', ..., '']`

This means that the `let operations` property was actually an array of operations and an empty string. By filtering this out properly and ensuring this is actually an array of operations (or conflict lines) we can apply the EOL improvement we need to make to the `writeTodoStorageFile` function